internal/host/hostcommands/install_cmd.go

```diff
@@ -303,8 +303,16 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 	}
 
 	for _, disk := range inventory.Disks {
-		if disk.DriveType == models.DriveTypeMultipath && disk.ID == host.InstallationDiskID {
-			installerArgs = append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
+		if disk.ID == host.InstallationDiskID {
+			if disk.DriveType == models.DriveTypeMultipath {
+				installerArgs = append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
+			} else if disk.DriveType == models.DriveTypeISCSI {
+				installerArgs = append(installerArgs, "--append-karg", "rd.iscsi.firmware=1")
```

Comment:
Since this PR looks like it's adding generic support for iSCSI on bare metal, I will mention here that auto-configuration via iBFT (used by `rd.iscsi.firmware`) is not always a given. The other way iSCSI configuration is provided is via `netroot=iscsi:...` (see [dracut.cmdline(7)](https://www.man7.org/linux/man-pages/man7/dracut.cmdline.7.html)).

If we want to provide a nice UX for it, we could provide separate fields for e.g. server name, target name, initiator name, etc... and assemble the `netroot` karg ourselves. This is similar to what blivet (Anaconda storage backend) does: https://github.com/storaged-project/blivet/blob/57f1382f19a1390dd2176e4f4772bab89380cfdf/blivet/devices/disk.py#L379-L394.

Alternatively, if the UX allows users to provide custom kargs, then we could skip this for now and users can always directly provide the `netroot` karg themselves as an escape hatch (and having `rd.iscsi.firmware=1` still hardcoded _shouldn't_ hurt, though this code could also auto-detect if one of the custom kargs is `netroot` and skip it I guess).

Reply:
Users can provides custom kargs to the installer though the API (https://github.com/openshift/assisted-service/blob/master/docs/user-guide/install-customization.md#set-the-installer-params). Maybe we can add a word about customizing iSCSI settings in our doc in the future.

Reply:
That is a good point about generic support.  I'll make this PR focused on OCI.  We can explore generic support in the future if there is demand for it.